controllers/gitrepository_controller.go

```diff
@@ -164,6 +164,12 @@ func (r *GitRepositoryReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		// Sets a reconcile ID to correlate logs from all suboperations.
 		WithValues("reconcileID", uuid.NewUUID())
 
+	if !managed.Enabled() {
+		return ctrl.Result{}, &serror.Stalling{
+			Reason: "Libgit2TransportNotEnabled",
+			Err:    errors.New("libgit2 managed transport not initialized"),
+		}
+	}
```

Comment:
Returning a stalling error here wouldn't do anything.
All the result and error translation happens as part of `SummarizeAndPatch()`. In order to take advantage of it, this should be put below that. That'd also automatically insert Stalled condition in the object status.